orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityStage.groovy

```diff
@@ -32,13 +43,81 @@ import org.springframework.stereotype.Component
  * - creating a new server group with min = desired ({@link CaptureSourceServerGroupCapacityTask})
  * - restoring min after the deploy has completed ({@link ApplySourceServerGroupCapacityTask})
  */
+@Slf4j
 @Component
 class ApplySourceServerGroupCapacityStage implements StageDefinitionBuilder {
+  @Autowired
+  private FeaturesService featuresService
+
+  @Autowired
+  DeleteEntityTagsStage deleteEntityTagsStage
+
+  @Autowired
+  OortService oortService
+
+  @Autowired
+  RetrySupport retrySupport
+
   @Override
   void taskGraph(Stage stage, TaskNode.Builder builder) {
     builder
       .withTask("restoreMinCapacity", ApplySourceServerGroupCapacityTask)
       .withTask("waitForCapacityMatch", MonitorKatoTask)
       .withTask("forceCacheRefresh", ServerGroupCacheForceRefreshTask)
   }
+
+  @Override
+  List<Stage> afterStages(@Nonnull Stage stage) {
+    try {
+      def taggingEnabled = featuresService.isStageAvailable("upsertEntityTags")
+      if (!taggingEnabled) {
+        return []
+      }
+
+      def entityTags = fetchEntityTags(oortService, retrySupport, stage)?.get(0)
+      if (!entityTags) {
+        return []
+      }
+
+      return [
+        newStage(
+          stage.execution,
+          deleteEntityTagsStage.type,
+          "Cleanup Server Group Tags",
+          [
+            id  : entityTags.id,
+            tags: Collections.singletonList(PINNED_CAPACITY_TAG)
+          ],
+          stage,
+          SyntheticStageOwner.STAGE_AFTER
+        )
+      ]
+    } catch (Exception e) {
+      log.error(
+        "Unable to determine whether server group is pinned (serverGroup: {}, account: {}, region: {})",
+        stage.context.serverGroupName,
+        stage.context.credentials,
+        getRegion(stage),
+        e
+      )
+
+      // any false negatives (pinned server groups that were not detected) will be caught by the RestorePinnedServerGroupsAgent
+      return []
```

Comment:
Looking to avoid any chance of failing the overall pipeline due to an error fetching entity tags.